lib/src/rules/directives_ordering.dart

```diff
@@ -83,10 +124,24 @@ class _Visitor extends SimpleAstVisitor {
 
   @override
   void visitCompilationUnit(CompilationUnit node) {
+    _checkDartImportGoFirst(node);
+    _checkPackageImportBeforeRelative(node);
+  }
+
+  void _checkDartImportGoFirst(CompilationUnit node) {
     node.directives
         .where(_isImportDirective)
         .skipWhile(_isDartImport)
         .where(_isDartImport)
         .forEach(rule._reportLintWithDartImportGoFirstMessage);
   }
+
+  void _checkPackageImportBeforeRelative(CompilationUnit node) {
+    node.directives
+        .where(_isImportDirective)
+        .where(_isNotDartImport)
+        .skipWhile(_isAbsoluteImport)
+        .where(_isPackageImport)
+        .forEach(rule._reportLintWithPackageImportBeforeRelativeMessage);
+  }
```

Comment:
While the guideline isn't worded this way, I believe the intent is that relative imports should be last. The current implementation is fine, but it might be interesting to consider inverting the logic and run it over the directives in reverse. More specifically, starting at the end, skip over relative imports until you get to a non-relative import, then flag any relative imports you find after that.

I suspect that this would produce fewer lints, and hence provide a better UX.

Of course, this is all moot if you go with my earlier suggestion and write the whole rule as a state machine.